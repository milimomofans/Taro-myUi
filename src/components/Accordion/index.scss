@mixin fontStyle($s:26rpx,$w:500,$c:rgba(51,51,51,1)){
    font-size:$s;
    font-weight:$w;
    color:$c;
}
@mixin flexBox($a:center,$j:initial){
    display: flex;
    justify-content: $j;
    align-items: $a;
}
@import "~taro-ui/dist/style/components/icon.scss";

._Accordion{
    position: relative;
    width: 480rpx;
    height:60rpx;
}
.options_main{
    width: 480rpx;
    height:360rpx;
    border:2rpx solid rgba(221,221,221,1);
    opacity:1;
    border-radius:8rpx;
    overflow: hidden;
    transition: height .3s;
    position: absolute;
    z-index:101;
}
.options_close{
    width: 480rpx;
    height:60rpx;
    border:2rpx solid rgba(221,221,221,1);
    opacity:1;
    border-radius:8rpx;
    overflow: hidden;
    transition: height .3s;
    position: absolute;
    z-index:101;
}
.open{
    transform: rotate(180deg);
    transition: all .3s;
}
.close{
    transform: rotate(0deg);
    transition:transform .3s;
}
.scroll-view{
    height:300rpx;
    position: relative;
    z-index:101;
    background: white;
}
.options_cur{
    width: 100%;
    height:60rpx;
    padding:20rpx 20rpx 14rpx 16rpx;
    box-sizing: border-box;
    @include flexBox(center,space-between);
    view:nth-of-type(1){
        @include fontStyle(28rpx, bold, rgba(51,51,51,1));
    }
}
.borderBottom{
    border-bottom: 2rpx solid rgba(221,221,221,1);
}