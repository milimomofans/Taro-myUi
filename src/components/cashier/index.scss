.cashier-wrapper {
  height: 100vh;
  background: #F4F4F4;

  &.white {
    background-color: #fff;
  }
}
