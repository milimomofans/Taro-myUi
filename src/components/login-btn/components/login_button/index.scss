.login-button-component-wrapper {
  margin-top: -36px;
  display: flex;
  flex-direction: column;
  align-items: center;

  .login-button-component-item {
    padding: 0;
    margin: 36px 0 0 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 590px;
    height: 100px;
    background: linear-gradient(135deg, rgba(72, 209, 150, 1) 0%, rgba(0, 203, 116, 1) 100%);
    border-radius: 50px;
    border: none;

    .login-button-component-icon {
      width: 40px;
      height: 32px;
    }

    .login-button-component-text {
      margin-left: 16px;
      color: #FEFFFF;
      font-size: 36px;
      line-height: 50px;
    }

    &.phone {
      border: 1px solid rgba(0, 203, 116, 1);
      background: #fff;

      .login-button-component-icon {
        width: 28px;
        height: 40px;
      }

      .login-button-component-text {
        color: #2FCB88;
      }
    }
  }
}
