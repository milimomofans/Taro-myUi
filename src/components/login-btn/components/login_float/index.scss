.login-float-wrapper {
  padding: 80px 80px 188px 80px;
  background-color: #fff;

  .login-float-title {
    padding-bottom: 80px;
    display: flex;
    flex-direction: column;
    align-items: center;

    .login-float-title-icon {
      width: 200px;
      height: 200px;
    }

    .login-float-title-title {
      margin-top: 36px;
      text-align: center;
      color: #606060;
      font-size: 34px;
      line-height: 32px;
    }
  }
}
