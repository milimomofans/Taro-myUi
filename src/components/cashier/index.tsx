/**
 * created by frank on 2020/03/12
 */

import Taro, {Component, Config} from '@tarojs/taro'
import {View} from '@tarojs/components'
import bind from "bind-decorator";
import classNames from 'classnames'
import $wulongPay from 'wulong-pay'
import {wxLog} from 'water-utils'
import debounce from 'lodash/debounce'
import CashierGoods from "./components/cashier-goods/cashier-goods";
import CashierSuccess from "./components/cashier-success/cashier-success";
import './index.scss'


interface Props {
  orderNo: string[]
  token: string
  onPayFail: () => void
  onPaySuccess: () => void
  env?: 'production' | 'development'
  customClass?: string
  customStyle?: any
}

interface State {
  orderData
  couponData
  loading: boolean
  paySuccess: boolean
}

class Cashier extends Component<Props, State> {
  static options = {
    addGlobalClass: true
  };

  constructor(props) {
    super(props)
  }

  state: State = {
    orderData: {},
    couponData: {},
    loading: true,
    paySuccess: false
  }

  async componentWillMount() {
    wxLog.info('打开收银台')
    wxLog.info(`当前环境 ${process.env.NODE_ENV}`)
    wxLog.info(`当前订单号 ${JSON.stringify(this.props.orderNo)}`)
    // 每次都会用设置第一个订单号为关键字
    const firstOrder = this.props.orderNo[0]
    wxLog.setFilterMsg(firstOrder)
    // 执行正常逻辑
    await Taro.setNavigationBarTitle({title: '收银台'})
    this.getOrderGoods()
  }

  config: Config = {
    navigationBarTitleText: '收银台'
  }

  baseUrl = process.env.NODE_ENV === 'production'
    ?'https://api-pro-out.trip.wlkst.com'
    :'http://service-0wal9i5k-1257188045.gz.apigw.tencentcs.com/release'

  header = {Authorization: this.props.token}

  appid = Taro.getAccountInfoSync().miniProgram.appId

  @bind
  async getOrderGoods() {
    // 请求加载的数据
    try {
      this.setState({loading: true})
      await Taro.showLoading({title: '加载中...'})
      const res = await Taro.request({
        url: this.baseUrl + '/service/user-order/order/pay/page',
        data: {order_nos: this.props.orderNo},
        header: this.header,
        method: 'POST'
      })
      if (!res.data || res.data.code !== 0){
        await Taro.showToast({title:res.data.message,icon:'none'})
        throw new Error(res.data || res)
      }
      const nowData = res.data.data
      this.setState({orderData: nowData})
      // 不要钱 直接唤起
      if (nowData.pay_fee === 0) {
        await this.onsubmitOrder()
      }
      Taro.hideLoading()
    } catch (e) {
      await Taro.showToast({title: '网络繁忙，请稍后再试', icon: 'none'})
      wxLog.error(`获取订单数据出错 ${JSON.stringify(e)}`)
      throw e
    } finally {
      this.setState({loading: false})
    }
  }

  @bind
  async getCoupon(trans_no) {
    await Taro.showLoading({title:'加载中...'})
    const {data: {data}} = await Taro.request({
      url: this.baseUrl + '/service/user-order/order/pay/show',
      data: {trans_no},
      header: this.header,
      method: 'POST'
    })
    this.setState({couponData: data})
    wxLog.info(`支付成功，获取优惠券 ${JSON.stringify(data)}`)
  }

  payOrderFn(payInfo) {
    wxLog.info(`调用米大师支付接口  ${payInfo}`)
    return new Promise((resolve, reject) => {
      $wulongPay('app05184f16acb89', 'midas', payInfo, async res => {
        const { resultMsg, resultCode } = res
        if (resultCode === 0) {
          wxLog.info(`支付成功！米大师返回成功数据 ${JSON.stringify(res)}`)
          // await Taro.showToast({ title: resultMsg, icon: 'success' })
          resolve()
        } else {
          reject(resultMsg)
          console.error(payInfo,'payInfo')
          wxLog.error(`支付失败，米大师返回失败数据 ${JSON.stringify(res)}`)
        }
      })
    })
  }

  @bind
  async pay() {
    const {
      data: {
        data: {pay}
      }
    } = await Taro.request({
      url: this.baseUrl + '/service/user-order/order/pay',
      data: { order_nos: this.props.orderNo, pay_type: 1, app_id: this.appid },
      header: this.header,
      method: 'POST'
    })
    const {out_trade_no, pay_info} = pay
    wxLog.info(`支付接口回调成功 ${JSON.stringify(pay)}`)
    if (this.state.orderData.pay_fee > 0) {
      // 支付
      await this.payOrderFn(pay_info)
      return out_trade_no
    }
    return out_trade_no
  }

  @bind
  async onsubmitOrder() {
    // 支付
    try {
      await Taro.showLoading({ title: '加载中...' })
      const transNo = await this.pay()
      await this.getCoupon(transNo)
      await Taro.setNavigationBarTitle({ title: '' })
      await Taro.setNavigationBarColor({
        frontColor: '#ffffff',
        backgroundColor: '#00CB74'
      })
      this.setState({ paySuccess: true })
      Taro.hideLoading()
    } catch (e) {
      const err = typeof e === 'string' ? e : '网络错误，请稍后再试'
      await Taro.showToast({ title: err, icon: 'none' })
      wxLog.error(`支付时发生错误，错误为 ${JSON.stringify(e)}`)
      // this.props.onPayFail()
      throw e
    }
  }

  submitOrder = debounce(this.onsubmitOrder,500)

  @bind
  paySuccess() {
    const {onPaySuccess} = this.props
    wxLog.info('用户调用业务传入的pay success 方法')
    if (onPaySuccess) {
      onPaySuccess()
    }
  }

  render(): any {
    const {orderData, loading, paySuccess, couponData} = this.state
    const {customClass, customStyle} = this.props
    const nowClass = classNames(['cashier-wrapper', customClass, {'white': paySuccess}])
    return (
      <View className={nowClass} style={customStyle}>
        {
          loading
            ?
            <View />
            :
            <View>
              {
                paySuccess
                  ?
                  <CashierSuccess payFee={orderData.pay_fee} couponData={couponData} onPaySuccess={this.paySuccess} />
                  :
                  <CashierGoods onPay={this.submitOrder} orderData={orderData} />
              }
            </View>
        }
      </View>
    )
  }
}

export default Cashier
