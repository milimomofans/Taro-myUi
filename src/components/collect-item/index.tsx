import Taro , { Component } from '@tarojs/taro';
import { View} from '@tarojs/components';
import ActivityCollectItem from './components/ActivityCollectItem/ActivityCollectItem';
import PlayCollectItem from './components/PlayCollectItem/PlayCollectItem'
import ScenicCollectItem from './components/ScenicCollectItem/ScenicCollectItem';
import TravleCollectItem from './components/TravleCollectItem/TravleCollectItem'

export default class CollectItem extends Component {

   config = {
       navigationBarTitleText: ''
  }

  state={}

  componentWillMount () {}
  componentDidMount () {}
  componentWillReceiveProps () {}
  componentWillUnmount () {}
  componentDidShow () {}
  componentDidHide () {}
  componentDidCatchError () {}
  componentDidNotFound () {}
  render() {
    return (
      <View>
          <ActivityCollectItem></ActivityCollectItem>
          <PlayCollectItem></PlayCollectItem>
          <ScenicCollectItem></ScenicCollectItem>
          <TravleCollectItem></TravleCollectItem>
      </View>
    );
  }
}
