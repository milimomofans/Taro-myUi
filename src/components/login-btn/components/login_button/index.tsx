import Taro from '@tarojs/taro'
import { Button, Image, View } from '@tarojs/components'
import { loginStore, loginSuccessInit, appid } from 'water-utils/lib/lib/loginMethod'
import {postBehavior} from 'wl-analytics/dist'
import wechatIcon from './images/wechatIcon.png'
import phoneIcon from './images/phoneIcon.png'
import loginRequest from '../../service'


const LoginButtonComponent = ({ onLoginSuccess, onLoginFail, source }) => {
  async function getPhone(res) {
    try {
      const userDeny =
        res.detail.errMsg === 'getPhoneNumber:fail user deny' ||
        res.detail.errMsg === 'getPhoneNumber:user deny' ||
        res.detail.errMsg === 'getPhoneNumber:fail:user deny'
      const { session_id } = loginStore.getNowSessionData()
      if (res.detail && !userDeny) {
        await Taro.showLoading({ title: '加载中...' })
        const { encryptedData, iv } = res.detail
        await loginRequest['login/postPhone']({
          session_id,
          iv,
          encrypt_data: encryptedData,
          appid
        })
        const result = await loginRequest['login/loginByWechat']({
          session_id,
          appid,
          source
        })
        loginSuccessInit(result)
        // 记录用户行为数据
        postBehavior({
          app_id: source,
          act_code: 'login'
        });
        // 登录完成之后调取loginBack
        if (onLoginSuccess) {
          onLoginSuccess()
        }
      } else {
        if (onLoginFail) {
          onLoginFail()
        }
      }
    } catch (error) {
      Taro.reportAnalytics('login', {
        error
      })
      console.error(error)
    } finally {
      Taro.hideLoading()
    }
  }

  return (
    <View className="login-button-component-wrapper">
      <Button
        className="login-button-component-item"
        open-type="getPhoneNumber"
        onGetPhoneNumber={getPhone}
      >
        <Image src={wechatIcon} className="login-button-component-icon" />
        <View className="login-button-component-text">微信授权登录</View>
      </Button>
      <View className="login-button-component-item phone">
        <Image src={phoneIcon} className="login-button-component-icon" />
        <View className="login-button-component-text" onClick={onLoginFail}>手机验证码登录</View>
      </View>
    </View>
  )
}

LoginButtonComponent.options = {
  addGlobalClass: true
}

LoginButtonComponent.defaultProps = {
  onLoginSuccess: () => {
  },
  onLoginFail: () => {
  }
}
export default LoginButtonComponent
