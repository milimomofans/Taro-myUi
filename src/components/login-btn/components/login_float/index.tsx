import Taro, { useEffect, useState } from '@tarojs/taro'
import { Image, View } from '@tarojs/components'
import LoginBtn from '../login_button/index'
import LoginPhone from '../login_phone/login_phone'
import titleIcon from './images/titleIcon.png'

const LoginFloat = ({ onLoginSuccess, isOpenFloat, source }) => {
  const [showPhone, setShowPhone] = useState<boolean>(false)
  useEffect(() => {
    if (!isOpenFloat) {
      setShowPhone(false)
    }
  }, [isOpenFloat, showPhone])

  const loginFail = () => {
    setShowPhone(true)
  }

  const loginSuccess = () => {
    setShowPhone(false)
    onLoginSuccess()
  }

  return (
    <View className='login-float-wrapper'>
      <View className='login-float-title'>
        <Image className='login-float-title-icon' src={titleIcon} />
        <View className='login-float-title-title'>一部手机游武隆</View>
      </View>
      {
        showPhone ?
          <LoginPhone onSuccess={loginSuccess} source={source} />
          :
          <LoginBtn onLoginFail={loginFail} onLoginSuccess={loginSuccess} source={source} />
      }
    </View>
  )
}

LoginFloat.options = {
  addGlobalClass: true
}

LoginFloat.defaultProps = {
  onLoginSuccess: () => {
  }
}

export default LoginFloat
