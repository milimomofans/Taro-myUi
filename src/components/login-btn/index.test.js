import Nerv, {findDOMNode} from 'nervjs'
import {View} from '@tarojs/components'
import {
  renderToString
} from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import WtLoginBtn from '../../../.temp/components/login-btn/index'

describe('WtLoginBtn Snap', () => {
  it('render initial WtLoginBtn', () => {
    const component = renderToString(
      < WtLoginBtn api={()=>{}}>
        <View className="test">登陆成功</View>
      </WtLoginBtn> )
    expect(component).toMatchSnapshot()
  })
  it('render success WtLoginBtn', () => {
    const component = renderToString(
      < WtLoginBtn api={()=>{}} loginSuccess>
        <View className="test">登陆成功</View>
      </WtLoginBtn>
       )
    expect(component).toMatchSnapshot()
  })
  })

  describe('WtLoginBtn Event', () => {
  it('WtTopBar onClick', () => {
    const onClick = jest.fn()

    const component = renderIntoDocument(
      < WtLoginBtn api={()=>{}} loginSuccess onNormalClick={onClick}>
        <View className="test">登陆成功</View>
      </WtLoginBtn>
    )
    const componentDom = findDOMNode(component, 'login-btn-container').querySelector('.test')
    Simulate.click(componentDom)
    expect(onClick).toBeCalled()
  })
})
