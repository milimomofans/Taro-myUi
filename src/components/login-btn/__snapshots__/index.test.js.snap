// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WtLoginBtn Snap render initial WtLoginBtn 1`] = `"<div class=\\"login-btn-container\\"><button class=\\"login-btn \\" open-type=\\"getPhoneNumber\\">undefined<div class=\\"test\\">登陆成功</div></button></div>"`;

exports[`WtLoginBtn Snap render success WtLoginBtn 1`] = `"<div class=\\"login-btn-container\\"><button class=\\"login-btn \\">undefined<div class=\\"test\\">登陆成功</div></button></div>"`;
